src/server.ts

```diff
@@ -0,0 +1,41 @@
+import config from "./config";
+import NotificationController from "./notification/notification.controller";
+import menash, { ConsumerMessage } from "menashmq";
+import { log, Severity } from "./utils/logger";
+import { ServerError } from "./utils/error";
+import { IPermission } from "./utils/permission.inteface";
+
+export default class Consumer {
+    public queue: string | undefined;
```

Comment:
When it can be undefined?

Reply:
When it wasn't initialized with queue yet.

Reply:
Why create a consumer without a queue? Why not put it in a constructor?

Reply:
To be more generic. For example, if I would like to create more than one server, I could just define a new queue for it to consume.
You are right about the constructor but I couldn't put async functions and wait for them to be resolved in the constructor so I created a function that operates as a constructor.

Reply:
I hope so...